benchmarks/distributed/rpc/parameter_server/trainer/hooks.py

```diff
@@ -0,0 +1,96 @@
+from utils import process_bucket_with_remote_server
+
+import torch
+import torch.distributed as c10d
+
+
+def allreduce_hook(state, bucket):
+    r"""
+    A ddp communication hook that uses the process_group allreduce implementation.
+    Args:
+        state (object): maintains state during the training process
+        bucket (GradBucket): gradient bucket
+    """
+    cref = state.cref
```

Comment:
(Apologize if I have asked this question before) What does "c" in "cref" stands for?

Reply:
It stands for class.